### PR TITLE
Bug #9670: Fixed PJAX redirect in IE

### LINKED ISSUE #9670: Redirect problems in Internet Explorer 11 (IE11)
When I use the included IE debugger and view the response headers, the header is there and the url is correct, but IE11 isn't reading it.

The response error:
pjax-response-error

The response headers:
pjax-response-headers

One comment on StackOverflow suggested using the status code `200`, and after testing, this does appear to work.  I know the default redirect status code is 302, but this just doesn't work.

## Update

I just respond with redirect to ajax request. Page redirects in all browsers except IE. I'm testting in IE11. I'm getting the following error in console: XMLHttpRequest: Network Error 0x2f76, Could not complete the operation due to error 00002f76..

URL for redirect sent in X-Redirect header. yii.js script handles this header, see line 265. var url = xhr.getResponseHeader('X-Redirect');. In case of IE11 getResponseHeader() returns null instead of URL. As I mentioned before I just respond with 200 status at the moment instead of 302.

### PR DESCRIPTION
**Pull request by SilverFire**
Closes #9670
Works in pair with https://github.com/yiisoft/jquery-pjax/pull/38

### CODE REVIEW

File: framework/web/Response.php [hunk 1]
```diff
@@ -758,7 +759,7 @@ public function redirect($url, $statusCode = 302, $checkAjax = true)
             $url = Yii::$app->getRequest()->getHostInfo() . $url;
         }
 
-        if ($checkAjax) {
+        if ($checkAjax && Yii::$app->getRequest()->getHeaders()->get('X-Ie-Redirect-Compatibility') !== null) {
```
Comment (klimov-paul) [inline framework/web/Response.php]: This missmatches the description:

> Takes effect only when request header `X-Ie-Redirect-Compatibility` is absent.

It should be:

```php
Yii::$app->getRequest()->getHeaders()->get('X-Ie-Redirect-Compatibility') === null
```

then

**Comment by klimov-paul** [comment]
This PR breaks regular AJAX redirection.
Perhaps the condition is incorrect, see
https://github.com/yiisoft/yii2/pull/10483/files#r49183315

File: framework/web/Response.php (hunk 1 above)
Comment (SilverFire) [inline framework/web/Response.php]: Thank you, you're right. Had a TODO to check this PR after holidays, but you came first :)
https://github.com/yiisoft/yii2/commit/019587e544f6909e7f6d2c83546fb2e063b801b9

**Comment by SilverFire** [comment]
Fixed

**Comment by klimov-paul** [comment]
Still fix seems to be incorrect.
With the fix AJAX will follow location __before__ response being processed by JS.
The better fix will be usage of the 200 status code in case of `X-Ie-Redirect-Compatibility` is sent.

```php
if ($checkAjax && ) {
    if (Yii::$app->getRequest()->getHeaders()->get('X-Ie-Redirect-Compatibility') === null && func_num_args() < 2) {
        $statusCode = 200;
    }
            if (Yii::$app->getRequest()->getIsPjax()) {
                $this->getHeaders()->set('X-Pjax-Url', $url);
            } elseif (Yii::$app->getRequest()->getIsAjax()) {
                $this->getHeaders()->set('X-Redirect', $url);
            } else {
                $this->getHeaders()->set('Location', $url);
            }
        } else {
            $this->getHeaders()->set('Location', $url);
        }
```

**Comment by klimov-paul** [comment]
See #10534

**Pull request merged**